components/HomeScreen/ContinentsList/ContinentsList.utils.js

```diff
@@ -2,16 +2,16 @@
  * creates a function to be called in the continent Pressable
  *
  * @param {object} args
- * @param {object} args.router
- * @param {function} args.router.push
+ * @param {object} args.navigation
+ * @param {function} args.navigation.navigate
  * @param {string} args.code
  */
 export function yieldDoNavigate(args) {
 
   const {
-    router,
+    navigation,
```

Comment:
why separate into a separate args value instead of just consuming it from the top?

Reply:
export function yieldDoNavigate({ navagation, code })

Reply:
One reason to do this would be if args was allowed to be undefined. Can't destructure undefined. That isn't the case here.

This is mirroring the coding style in this project's components, where the props are destructured just like this. This is just a coding style that I like. Seems there are pros and cons to both ways, but they're just about the same. I'm willing to do the way you've shown in our real projects.